ext/standard/uuid.c

```diff
@@ -0,0 +1,662 @@
+/*
+  +----------------------------------------------------------------------+
+  | PHP Version 7                                                        |
+  +----------------------------------------------------------------------+
+  | Copyright (c) 1997-2017 The PHP Group                                |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | http://www.php.net/license/3_01.txt                                  |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Richard Fussenegger <php@fleshgrinder.com>                   |
+  +----------------------------------------------------------------------+
+*/
+
+/* $Id$ */
+
+#include "php_uuid.h"
+
+#include "zend_exceptions.h"
+#include "ext/spl/spl_exceptions.h"
+#include "md5.h"
+#include "php_random.h"
+#include "sha1.h"
+
+PHPAPI zend_class_entry *php_ce_UUID;
+PHPAPI zend_class_entry *php_ce_UUIDParseException;
+
+static const uint8_t UUID_VERSION_MIN = 0;
+static const uint8_t UUID_VERSION_MAX = 15;
+
+static const uint8_t UUID_HEX_LEN = sizeof(php_uuid_hex) - 1;
+static const uint8_t UUID_STRING_LEN = sizeof(php_uuid_string) - 1;
+
+static const char UUID_BYTES_PROP[] = "bytes";
+static const uint8_t UUID_BYTES_PROP_LEN = sizeof(UUID_BYTES_PROP) - 1;
```

Comment:
Even though all of these are small, they should be declared as `size_t`, as they are string lengths.

Reply:
(Unrelated comment) Given these are properties, it would make more sense to me to cache these as `zend_string`s, given that's what they'll have to be converted to otherwise.